Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/DriverOnboarding/VehicleRegistrationCertificate.hs

```diff
@@ -118,26 +120,7 @@ verifyRC isDashboard mbMerchant (personId, _) req@DriverRCReq {..} = do
           unless (extractRCNumber == rcNumber) $
             throwImageError imageId $ ImageDocumentNumberMismatch (maybe "null" maskText extractRCNumber) (maybe "null" maskText rcNumber)
         Nothing -> throwImageError imageId ImageExtractionFailed
-
-  now <- getCurrentTime
-  mDriverAssociation <- DAQuery.getActiveAssociationByDriver personId
-
-  case mDriverAssociation of
-    Just driverAssociaion -> do
-      driverRC <- RCQuery.findById driverAssociaion.rcId >>= fromMaybeM (InvalidRequest "Missing RC entry")
-      rcNumber <- decrypt driverRC.certificateNumber
-      unless (rcNumber == vehicleRegistrationCertNumber) $ throwImageError imageId DriverAlreadyLinked
-      unless (driverRC.fitnessExpiry < now) $ throwImageError imageId RCAlreadyUpdated -- RC not expired
```

Comment:
For this we should still check if driver is already linked to same RC, right?

Reply:
if RC is already linked then requirement was RC should be linked but not active, to make it active we will follow another flow, so according to me here we shouldn't, but while making it active/inactive we check this.